### PR TITLE
Hypermedia

### PR DESCRIPTION
**Pull request by TNagaraja**
Issue #5

### CODE REVIEW

File: wfl/Profiles/RootProfile.cs
Comment (jspringer2020) [inline wfl/Profiles/RootProfile.cs:30]: Just out of curiousity, why is the relation different than the representation? Is the representation the expected output type being returned from the link?

File: wfl/App_Start/WebApiConfig.cs
Comment (jspringer2020) [inline wfl/App_Start/WebApiConfig.cs:15]: I haven't looked at the actual Hypermedia.NET project in a while, but what does the `ReflectiveTransform()` do for us here?

File: wfl/Profiles/RootProfile.cs
Comment (jspringer2020) [inline wfl/Profiles/RootProfile.cs:7]: Personal opinion here, I'd suggest creating a child namespace of `Hypermedia`. In other projects we've also had mapping profiles, validation profiles, etc.  That too gets complicated, so any thoughts on putting all the files for a "type" in the same place -- for example, having Root, RootProfile, RootController, etc all in one spot (this is a discussion question by the way)?

Comment (TNagaraja) [reply wfl/Profiles/RootProfile.cs:7]: Why would the controllers be in a `Hypermedia` child namespace?

File: wfl/Profiles/RootProfile.cs
Comment (TNagaraja) [reply wfl/Profiles/RootProfile.cs:30]: Yes, I was operating under that assumption.

Comment (JChapman202) [reply wfl/Profiles/RootProfile.cs:30]: A relation describes how a given item relates to the item it is on.  A representation is API defined and can be used to inform a consumer regarding expected API behavior (class property in Siren).  In this case I think it makes sense to represent the type of class to expect when you follow the link.  We could also choose to just leave this class off as well though.

File: wfl/App_Start/WebApiConfig.cs
Comment (TNagaraja) [reply wfl/App_Start/WebApiConfig.cs:15]: To be upfront, I don't fully understand what `ReflectiveTransform()` is doing here. I was modeling this set up based upon the Hypermedia sample.

Comment (JChapman202) [reply wfl/App_Start/WebApiConfig.cs:15]: If there is no entity specific transform selected for an entity, it'll use the reflective transform.  The reflective transform simply looks over all public properties of an object and returns them as properties.  It then also adds a class based on the class name I believe.  I'm guessing this was added mostly as a "just in case" we don't have a profile registered?

File: wfl/Profiles/RootProfile.cs
Comment (JChapman202) [reply wfl/Profiles/RootProfile.cs:7]: I agree this wouldn't be part of a Hypermedia namespace.  We could choose to have a FoodOrder namespace and Restaurant namespace etc. to have more fine grained folders.  I'm also fine if we delay that until we grow larger.  Personally I don't yet have an issue with what we have so far.

**Review by JChapman202 — CHANGES_REQUESTED** [review]
Great stuff Tej.  This is definitely going to make us get off our asses and make Hypermedia public.

However, before we merge can you create a profile for the restaurant object?  Then include then menu link as a hypermedia link on that object?

**Comment by TNagaraja** [comment]
Added Restaurant and Food Order Profiles

**Review by enricoferrari — APPROVED** [review]

**Review by jspringer2020 — APPROVED** [review]
Forgot to approve when I reviewed this originally.

**Review by JChapman202 — APPROVED** [review]

**Pull request merged**
